static/js/stream_data.js

```diff
@@ -46,6 +47,27 @@ exports.get_colors = function () {
     return _.pluck(exports.subscribed_subs(), 'color');
 };
 
+exports.get_subscribers_count = function (stream_name) {
+    var sub = exports.get_sub(stream_name);
+    if (typeof sub === 'undefined') {
+        return undefined;
+    }
+    var count = Object.keys(sub.subscribers._items).length;
+
+    if (sub.subscribed === false && sub.subscribers.has(page_params.email) === true) {
+        // If user is not subscribed (i.e. sub.subscribed is false)
```

Comment:
In existing code, if user unsubscribes himself from a stream, then for that stream `sub.subscribed` is changed to `false`. But, user still exists in subscribers list till windows is refreshed. This might be a possible bug.
In case of someone else unsubscribing from stream, the user is removed from subscriber list.